hack/update-licenses.sh

```diff
@@ -0,0 +1,186 @@
+  if [[ -z "${CONTENT[${index}]-}" ]]; then
+    for f in ${local_files[@]-}; do
+      # Find some copyright info in any file and break
+      if egrep -i -wq "${ensure_pattern}" "${f}"; then
+        CONTENT[${index}]="${f}"
+        break
+      fi
+    done
+  fi
+}
+
+
+#############################################################################
+# MAIN
+#############################################################################
+
+# Check bash version
+if ((${BASH_VERSINFO[0]}<4)); then
+  echo
+  echo "ERROR: Bash v4+ required."
+  # Extra help for OSX
+  if [[ "$(uname -s)" == "Darwin" ]]; then
+    echo
+    echo "Ensure you are up to date on the following packages:"
+    echo "$ brew install md5sha1sum bash"
+  fi
+  echo
+fi
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+LICENSES_FILE="${REPO_ROOT}/LICENSES"
+TMP_LICENSE_FILE="/tmp/cluster-registry.LICENSES.$$"
+DEPS_DIR="vendor"
+
+cd "${REPO_ROOT}"
+declare -Ag CONTENT
+
+# Put the cluster-registry LICENSE on top
+(
+echo "================================================================================"
+echo "= cluster-registry licensed under: ="
+echo
+cat "${REPO_ROOT}/LICENSE"
+echo
+echo "= LICENSE $(cat "${REPO_ROOT}/LICENSE" | md5sum | awk '{print $1}')"
+echo "================================================================================"
+) > "${TMP_LICENSE_FILE}"
+
+# Loop through every package in vendor.
+for PACKAGE in $(dep status | \
+                 tail -n +2 |
+                 awk '{print $1}' |
+                 sort -f); do
+  process_content ${PACKAGE} LICENSE
+  process_content ${PACKAGE} COPYRIGHT
+
+  # display content
+  echo
+  echo "================================================================================"
+  echo "= ${DEPS_DIR}/${PACKAGE} licensed under: ="
+  echo
+
+  file=""
+  if [[ -n "${CONTENT[${PACKAGE}-LICENSE]-}" ]]; then
+      file="${CONTENT[${PACKAGE}-LICENSE]-}"
+  elif [[ -n "${CONTENT[${PACKAGE}-COPYRIGHT]-}" ]]; then
+      file="${CONTENT[${PACKAGE}-COPYRIGHT]-}"
+  fi
+  if [[ -z "${file}" ]]; then
+      cat > /dev/stderr << __EOF__
+No license could be found for ${PACKAGE} - aborting.
+
+Options:
+1. Check if the upstream repository has a newer version with LICENSE and/or
+   COPYRIGHT files.
+2. Contact the author of the package to ensure there is a LICENSE and/or
+   COPYRIGHT file present.
+3. Do not use this package in the cluster-registry.
+__EOF__
+      exit 9
+  fi
+  cat "${file}"
+
+  echo
+  echo "= ${file} $(cat ${file} | md5sum | awk '{print $1}')"
+  echo "================================================================================"
+  echo
+done >> "${TMP_LICENSE_FILE}"
+
+if [[ ${1:-} == "--verify-only" ]]; then
+  if ! _out="$(diff -Naupr "${LICENSES_FILE}" "${TMP_LICENSE_FILE}")"; then
+    echo "Your LICENSES file is out of date. Run hack/update-licenses.sh and commit the results."
+    echo "${_out}"
+    exit 1
+  fi
+  exit 0
+fi
+
+cat "${TMP_LICENSE_FILE}" > "${LICENSES_FILE}"
```

Comment:
We need to cleanup the `${TMP_LICENSE_FILE}` after we're done. How about we just `mv -f ${TMP_LICENSE_FILE} ${LICENSES_FILE}`?

Reply:
SGTM.